LUN-256: Added QR code scanner for ios

f1sh1918 opened the pull request:
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

Tested on ios and android

klinzo commented on ios/Lunes/Info.plist:
```diff
@@ -2,6 +2,8 @@
 <plist version="1.0">
 <dict>
+    <key>NSCameraUsageDescription</key>
+    <string>Um Barcodes zu scannen, benötigt LUNES Zugriff auf ihre Kamera</string>
```
```suggestion
    <string>Zum Scannen von QR-Codes benötigt LUNES Zugriff auf Ihre Kamera.</string>
```

klinzo commented on src/constants/labels.json:
```diff
@@ -22,7 +22,7 @@
     "qrCodeScanner": {
       "noAuthorization": {
-        "description": "Kamerazugriff zum Scannen von QR-Codes erforderlich. Öffnen Sie die Android-Einstellungen, tippen Sie auf Berechtigungen und tippen Sie auf Erlauben.",
+        "description": "Kamerazugriff zum Scannen von QR-Codes erforderlich. Öffnen Sie die Einstellungen, tippen Sie auf Berechtigungen und tippen Sie auf Erlauben.",
```
I guess this only applies for android, not sure how you can grant permissions in iOS.

f1sh1918 replied:
i mean we could adjust these text but there is a `Einstellungen` Button below the text and then you reach the settings section on each platform

klinzo replied:
Even though this is incorrect on iOS, it is fine for me. There is no such thing as `Berechtigungen` or `Erlauben`. You just toggle the switch `Kamera`. If you open the settings directly and not from the button in lunes, you just need to select `Lunes` and toggle the switch.

SteffiStuffel commented on ios/Lunes/Info.plist:
```diff
@@ -2,6 +2,8 @@
 <plist version="1.0">
 <dict>
+    <key>NSCameraUsageDescription</key>
+    <string>Zum Scannen von QR-Codes benötigt LUNES Zugriff auf Ihre Kamera.</string>
```
Lunes uses informal form of adress. So should be `deine Kamera`

SteffiStuffel reviewed (approved):
Tested on browserstack ios: Is not working, but may no be due to you changes, but because of browserstack.
Tested on android real devices: still works fine.

f1sh1918 commented:
> Tested on browserstack ios: Is not working, but may no be due to you changes, but because of browserstack. Tested on android real devices: still works fine.

works on real ios devices. browserstack also crashes for me

f1sh1918 commented:
> Tested on iOS simulator, permissions are handled correctly. I could not scan a QR code obviously but I think that should work just fine. At least on the simulator (iPhone 12 Pro Max) the close button on the QR code scanner is cut and not pressable/not working.

fixed this by adding `SafeAreaView` and adjusted Icon positioning
